microsoft/Windows-appsample-customers-orders-database
# feat (bump target versions): [general] update to WinUI 3

## WHY

we want to refresh the well-known `Contoso.App` to start using modern libraries and SDKs such us `.NET6`, `EF6` and `WinUI3` by removing all `UWP` dependencies where possible without introducing code changes from this PR but only those are strictly required to get this working. 

Additionally we want to upgrade to the latest `MSAL` patterns recommendations to acquire tokens silently and interactively (only if needed) as well as follow the modern recommended way of caching token for Desktop apps.

## What Changed

- `Models`, `Repository` and `Service` projects were bumped to `.NET 6`
- Migrate `Contoso.App` to WinUI 3
- Migrate Contoso Db based on EF breaking changes
- Remove previous UWP identity implementation ([Web Account Manager](https://docs.microsoft.com/en-us/windows/uwp/security/web-account-manager))
- Implement new MSAL experience
- Implement sign-out flow by removing cached tokens. 
- Remove previous Windows.Storage.ApplicationData in favor of retrieving the cached tokens.
- Implement recommended way for token caching in desktop apps.
- Update README

How to Test?

As a user you are now being able to use the `Contoso.App` as a valid legacy `UWP` app that has been recently migrated `WinUI 3` by having in mind that there are bunch of improvements yet to be done.

You can follow te README to execute the `Contoso.App` from Microsoft Visual Studio 2022 version `17.01` or later, and it is working as expected without UI style changes.

You configure your Azure AD Application from repository `Constants`, and you can now sign-in/out successfully into Azure Tenant Organization using OAuth 2.0 client credentials flow on top of the Microsoft Identity platform.

If you signed-in and restarted the application you are now being automatically signed-in without being asked to enter your credentials. As part of this process your token is being acquired silently.
